sqlitecluster/SQLiteNode.cpp

```diff
@@ -1652,8 +1652,9 @@ void SQLiteNode::_onMESSAGE(SQLitePeer* peer, const SData& message) {
                     // and waiting for the transaction that failed will be stuck in an infinite loop. To prevent that
                     // we're changing the state to SEARCHING and sending the cancelAfter property to drop all threads
                     // that depend on the transaction that failed to be threaded.
-                    _changeState(SQLiteNodeState::SEARCHING, message.calcU64("NewCount") - 1);
+                    _replicationThreadCount.fetch_sub(1);
                     SWARN("Caught system_error starting _replicate thread with " << _replicationThreadCount.load() << " threads. e.what()=" << e.what());
+                    _changeState(SQLiteNodeState::SEARCHING, message.calcU64("NewCount") - 1);
                     STHROW("Error starting replicate thread so giving up and reconnecting.");
```

Comment:
What happens after we throw here?

Reply:
We should go synchronizing as if the node was just turned on.